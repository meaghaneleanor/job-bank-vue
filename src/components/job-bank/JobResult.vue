<template>
  <div class="ontario-job-bank__single-job">
    <div class="grid-x">
      <NewJobFlag />
      <div class="ontario-job-bank-result__single-job--logo ontario-grid-cell small-2">
        Logo
      </div>
      <div class="ontario-job-bank-result__single-job--info  ontario-grid-cell large-8 small-7">
        <p>Title</p>
        <p>Company</p>
        <p>Location</p>
        <p>$$ Salary $$</p>
      </div>
      <div class="ontario-job-bank-result__single-job--job-details-button  ontario-grid-cell large-2 small-3">
        <a href="#" class="ontario-button ontario-button--primary">
          Job details
        </a>
      </div>
    </div>
  </div>
</template>

<script>
import NewJobFlag from './NewJobFlag.vue';

export default {
  components: {
    NewJobFlag
  }
};
</script>
