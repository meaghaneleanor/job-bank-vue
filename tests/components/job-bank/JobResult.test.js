import { shallowMount } from '@vue/test-utils';
import { $t } from '../../test.utils.js';
import $i18n from '../../test.utils.js';

import JobResult from '../../../src/components/job-bank/JobResult.vue';

const defaultOptions = {
  mocks: {
    $t, 
    $i18n
  },
  propsData:{
    job: {
      title: 'Cool Guy',
      company: 'ODS',
      city: 'Toronto',
      salary: '$1000 per hour',
      url: 'www.ontario.ca',
      date_posted: '2019-07-10T10:59:00Z'
    }
  },
  stubs: ['NewFlag']
};

const createComponent = shallowMount(JobResult, defaultOptions);

describe('JobResult.vue', () => {
  // eslint-disable-next-line no-unused-vars
  let wrapper;
      
  beforeEach(() => {
    wrapper = createComponent;
  });
  
  describe('snapshot', () => {
    it('should display the expected html', () => {
      let wrapper = createComponent;
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('.addSalaryString', () => {
      describe('When the salary data contains a $', () => {
        it('should return false', () => {
          let mock = '$100 per hour';
          let salaryData = wrapper.vm.addSalaryString(mock);
          expect(salaryData).toEqual(false);
        });
      });

      describe('When the salary data does not contain a $', () => {
        it('should return true', () => {
          let mock = '100 per day (to be negotiated)';
          let salaryData = wrapper.vm.addSalaryString(mock);
          expect(salaryData).toEqual(true);
        });
      });
    });
  });
});
