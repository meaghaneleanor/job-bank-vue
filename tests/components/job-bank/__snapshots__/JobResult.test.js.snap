// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobResult.vue snapshot should display the expected html 1`] = `
"<div class=\\"ontario-job-bank__single-job\\"><div class=\\"grid-x\\"><newflag-stub date=\\"2019-07-10T10:59:00Z\\"></newflag-stub> <div class=\\"ontario-job-bank-result__single-job--logo ontario-grid-cell large-1 medium-1 small-2\\"><img src=\\"/jobResultLogo.png\\" alt=\\"Ontario Trillium Logo\\"></div> <div class=\\"ontario-job-bank-result__single-job--info  ontario-grid-cell large-8 medium-8 small-7\\"><p><a href=\\"www.ontario.ca\\">Cool Guy</a></p> <p> </p> <p>, ON</p> <p><!---->
        $1000 per hour
      </p></div> <div class=\\"ontario-job-bank-result__single-job--job-details-button ontario-grid-cell large-3 medium-3 small-3\\"><a href=\\"www.ontario.ca\\" class=\\"ontario-button ontario-button--primary\\">
        jobPage.detailsButton
      </a></div></div></div>"
`;
